src/LibraryManager/Providers/jsDelivr/JsDelivrCatalog.cs

```diff
@@ -185,14 +189,16 @@ public async Task<CompletionSet> GetLibraryCompletionSetAsync(string libraryName
                         return completionSet;
                     }
 
-                    IEnumerable<string> packageNames = await NpmPackageSearch.GetPackageNamesAsync(libraryNameStart, CancellationToken.None);
+                    IEnumerable<string> packageNames = await _packageSearch.GetPackageNamesAsync(libraryNameStart, CancellationToken.None);
 
                     foreach (string packageName in packageNames)
                     {
+                        NpmPackageInfo packageInfo = await _packageInfoCache.GetPackageInfoAsync(packageName, CancellationToken.None);
+
                         var completionItem = new CompletionItem
```

Comment:
This used to do one request for package names, now it's going to do lots of requests for the info for every package. Is there a way InsertionText can be computed on-demand? The majority of InsertionText values that are created are going to be discarded, so ideally we would save those requests until they're actually necessary.

(This applies to the lines above. Apparently I can't aim GitHub comments.)

Reply:
I think it's possible now that we're using the NPM registry for the name queries.  The response body we get back includes the latest version; the search API we used prior to #494 didn't include any version info.  We'll have to change where we parse the response info to do this, but it should now be possible without multiple requests.

Reply:
Fixed with commit 7. NpmPackageSearch methods will now return NpmPackageInfo rather than just a package name, which I take advantage of here